util/imagetools/loader.go

```diff
@@ -336,6 +358,23 @@ func (l *loader) scanProvenance(ctx context.Context, fetcher remotes.Fetcher, r
 					if err != nil {
 						return nil, err
 					}
+
+					if layer.MediaType == "application/vnd.in-toto.provenance+dsse" {
```

Comment:
Could we get this part into a shared function that is called by both provenance and SBOM code paths?

Also, define these mediatypes that are used multiple times as (private) const on top of the file.